src/lj_str.c

```diff
@@ -130,49 +130,6 @@ void lj_str_resize(lua_State *L, MSize newmask)
   g->strhash = newhash;
 }
 
-static MSize
-lj_str_original_hash(const char *str, size_t lenx)
```

Comment:
This should stay here and not go into lj_str_hash.c.  Only then will it be safe to build lj_str_hash.c with crc32 support enabled, keeping everything else under default configuration so that it is widely deployable without rebuilding.

Reply:
`lj_str_hash.c` is safe to be built with or without CRC32 support, and to be loaded at runtime without rebuilding. Let's not forget that `lj_str_hash_orig` does not exist in LuaJIT's upstream v2.1 branch, and that it was introduced when the CRC32 hashing optimization was added in https://github.com/openresty/luajit2/commit/7923c6343c26eb6dd3ffcc8709ff6783ff0ff80f.
Where this function lives makes no difference, and maintaining it in `lj_str_hash.c` allows to group all string hashing implementations (including future ones, if any) and reduces cognitive load for readers and maintainers.